src/realm/error_codes.hpp

```diff
@@ -36,6 +36,10 @@ class ErrorCodes {
         LogicError = 3,
         BrokenPromise = 4,
         OperationAborted = 5,
+        ReadError = 7,
```

Comment:
Are these going to conflict with exception unification later on?

Reply:
It will unfortunately. We may need to add additional error codes when the server starts sending back more close codes, so I think there is no way around it.

Reply:
It looks like they will:
https://github.com/realm/realm-core/blob/feature/exception-unification/src/realm/error_codes.h#L185

We may want to create a function that converts our current websocket status codes (1000-1015, 4000-4999) into exception unification compatible `ErrorCodes::Error` values and add any new values that we need.

Reply:
Yes, we'll probably need that